src/fileparse.py

```diff
@@ -0,0 +1,66 @@
+import csv
+import logging
+from pathlib import Path
+from typing import Any, List, TextIO, Tuple, Union, Callable, Dict, cast
+
+
+def parse_csv(
+    file_path_or_buffer: Union[str, Path, TextIO],
+    select: List[str] = None,
+    types: List[Callable] = None,
```

Comment:
Just think of this but this argument is not restricted to types: any callable can be used! For example, if I want to read a column, named `my_float`, with float and keep only 2 digits I can do:
```python
from functools import partial

round_2_digit_fn = partial(round, ndigits=2)
parse_csv(my_file_path, select=["my_float"], types=[round_2_digit_fn])
```

This will read the CSV and apply the `round_2_digit_fn` to each element of the column `my_float`.

In this case it is not really a type it something like a function to convert the elements. So we can rename `types` to something like `convert_fn` to make it more generic, does it make sense?

Reply:
Yes. I agree as well.